.github/actions/test-setup/action.yml

```diff
@@ -1,56 +1,99 @@
-    - name: Edit environment variables and credentials
+
+    - name: Configure Protected node
+      working-directory: protected_node/recipes
       run: |
-        # Replace NB_API_QUERY_URL placeholder
-        sed -i 's|^NB_API_QUERY_URL=.*|NB_API_QUERY_URL=http://localhost:8080|g' .env
-        sed -i 's|^NB_FEDERATE_REMOTE_PUBLIC_NODES=.*|NB_FEDERATE_REMOTE_PUBLIC_NODES=false|g' .env
+        cp template.env .env
+        cp local_nb_nodes.template.json local_nb_nodes.json
+
+        # Replace defaults, even if they are commented out
+        sed -i 's|^#\?[[:space:]]\?COMPOSE_PROJECT_NAME=.*|COMPOSE_PROJECT_NAME=protected_node|' .env
+        sed -i 's|^#\?[[:space:]]\?LOCAL_GRAPH_DATA=.*|LOCAL_GRAPH_DATA=../openneuro_mini_data|' .env
+        sed -i 's|^#\?[[:space:]]\?NB_FEDERATE_REMOTE_PUBLIC_NODES=.*|NB_FEDERATE_REMOTE_PUBLIC_NODES=false|' .env
+        sed -i 's|^#\?[[:space:]]\?NB_API_QUERY_URL=.*|NB_API_QUERY_URL=http://localhost:8080|' .env
 
         # Replace default username/password
-        echo "testadminpassword" > secrets/NB_GRAPH_ADMIN_PASSWORD.txt
-        echo "testpassword" > secrets/NB_GRAPH_PASSWORD.txt
+        echo "testadminpassword1" > secrets/NB_GRAPH_ADMIN_PASSWORD.txt
+        echo "testpassword1" > secrets/NB_GRAPH_PASSWORD.txt
+
+        # Define local nodes to federate over
+        echo '[{"NodeName": "2 OpenNeuro Datasets", "ApiURL": "http://protected_node-api-1:8000"},{"NodeName": "BIDS Synthetic", "ApiURL": "http://open_node-api-1:8000"}]' > local_nb_nodes.json
       shell: bash
-    
-    - name: Launch services
+
+    - name: Launch Protected node
+      working-directory: protected_node/recipes
       run: |
         docker compose --profile full_stack up -d
+
+    - name: Checkout recipes for Open node (local_node)
+      uses: actions/checkout@v4
+      path: open_node
+
+    - name: Configure Open node
+      working-directory: open_node/recipes
+      run: |
+        cp template.env .env
+
+        sed -i 's|^#\?[[:space:]]\?COMPOSE_PROJECT_NAME=.*|COMPOSE_PROJECT_NAME=open_node|' .env
+        sed -i 's|^#\?[[:space:]]\?NB_GRAPH_PORT_HOST=.*|NB_GRAPH_PORT_HOST=7201|' .env
+        sed -i 's|^#\?[[:space:]]\?NB_NAPI_PORT_HOST=.*|NB_NAPI_PORT_HOST=8001|' .env
+        sed -i 's|^#\?[[:space:]]\?NB_RETURN_AGG=.*|NB_RETURN_AGG=false|' .env
+
+    - name: Launch Open node
+      working-directory: open_node/recipes
+      run: |
+        docker compose --profile local_node up -d
       shell: bash
 
-    - name: Wait for graph setup completion
-      working-directory: ./scripts
+    - name: Wait for graph setup completion for both nodes
       run: |
-        # The log is written at the end of the setup, so we must wait for the file to exist
-        while [ ! -f ./logs/DEPLOY.log ]; do
-          :
+        until [ -f protected_node/recipes/scripts/logs/DEPLOY.log ] && grep -q "Finished setting up the Neurobagel graph backend." protected_node/recipes/scripts/logs/DEPLOY.log; do
+          sleep 1
```

Comment:
why sleep here over just doing nothgin here?

Reply:
`sleep 1` reduces unnecessary checks by slowing down the iterations a bit, otherwise with `:` the loop will just continuously run without pause